### PR TITLE
clean up code and add jsdocs to each method

### PR DESCRIPTION
**Pull request by cvega**

### CODE REVIEW

File: test/frisby_httpbin.js [hunk 1]
```diff
@@ -1,3 +1,22 @@
+Skip to content
```
Comment (MarkHerhold) [inline test/frisby_httpbin.js]: @cvega Merge conflict?
CC @paulmelnikow

**Comment by MarkHerhold** [comment]
This looks like a really great PR! In the future, let's allow a brief period for reviews with other contributors if possible. Thanks!

File: lib/icedfrisby.js [hunk 2]
```diff
@@ -1,37 +1,34 @@
...
+
+const _ = require('lodash');
+const chai = require('chai');
+const chalk = require('chalk');
+const https = require('https');
+const pkg = require('../package.json');
+const pm = require('./pathMatch');
+const qs = require('qs');
+const request = require('request');
+const stream = require('stream').Stream;
```
Comment (paulmelnikow) [inline lib/icedfrisby.js]: The capital S in `Stream` signifies that it's a constructor function. It's a Crockford convention, and matches the module's convention. It makes more sense to keep the identifier capitalized.

File: lib/icedfrisby.js [hunk 3]
```diff
@@ -1,37 +1,34 @@
...
+const util = require('util');
+
+
+// setup defaults
+let _icedfrisbyDefaults = _.constant({
```
Comment (paulmelnikow) [inline lib/icedfrisby.js]: A bunch of the `let`s in this file could be `const`. I prefer to use `const` for identifiers which are not reassigned, as it conveys a lot more to the programmer. By making the `let`s stand out, this also encourages writing immutable code, which tends to be much easier to follow.

File: lib/icedfrisby.js [hunk 4]
```diff
@@ -125,131 +189,105 @@ function Frisby(msg) {
...
-// @param int Timeout in seconds
-//
-Frisby.prototype.timeout = function(t) {
-  if(!t) {
+/**
+ * @param {number} ms - timeout value in milliseconds
+ * @return {object}
+ * @desc Timeout getter and setter
+ */
+Frisby.prototype.timeout = function(ms) {
```
Comment (paulmelnikow) [inline lib/icedfrisby.js]: Love this.

File: lib/icedfrisby.js [hunk 5]
```diff
@@ -260,63 +298,117 @@ Frisby.prototype.auth = function(user, pass, digest) {
...
+ * Perform GET request against resource
+ * @param {string} uri - resource
+ * @param {object} [params] - uri parameters
+ * @return {object}
+ * @desc Perform an HTTP GET request
+ * @example
+ * .get('/resource', {key: value})
+ */
+Frisby.prototype.get = function (uri, params) {
+  return this._request.apply(this, ['GET', uri, null, params]);
```
Comment (paulmelnikow) [inline lib/icedfrisby.js]: 👍

File: lib/icedfrisby.js [hunk 6]
```diff
@@ -686,11 +855,15 @@ Frisby.prototype.inspectHeaders = function(message) {
...
+
+/**
+ * @param {string} message - message to print before the inspection
+ * @return {object}
+ * @desc Debugging helper to inspect the HTTP response body content
+ */
+Frisby.prototype.inspectBody = function (message) {
   this.inspect(
-    function(err, req, res, body) {
+    function (err, req, res, body) {
```
Comment (paulmelnikow) [inline lib/icedfrisby.js]: In the future, we could think about using arrow functions.

**Pull request merged**